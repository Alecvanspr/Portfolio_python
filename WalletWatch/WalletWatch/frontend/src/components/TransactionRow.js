import React, { Component } from "react";

export default class TransactionRow extends Component{
    constructor(props){
        super (props)
        this.state = {

        }
    }
      //geeft de kleur van het uitgegeven bedrag
    getTextColor = (type) => {
        if(type === "Inkomsten"){
        return "Inkomsten-text"
        }
        return "Uitgaven-text"
    }

    render(){
        const {item , editItem , handleDelete} = this.props
        return (
        <tr>
            <td></td>
          <td>{item.naam}</td>
          <td>{item.datum}</td>
          <td className={this.getTextColor(item.type)}>â‚¬{Math.round(item.bedrag)}</td>
          <td><button
            className="btn btn-secondary mr-2"
            onClick={()=> editItem(item)}
          >
            Edit
          </button></td>
          <td><button
            className="btn btn-danger"
            onClick={() => handleDelete(item)}
          >
            Delete
          </button></td>
        </tr>
        )
    }
}