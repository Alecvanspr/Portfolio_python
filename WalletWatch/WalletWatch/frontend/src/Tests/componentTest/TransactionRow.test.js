import renderer from 'react-test-renderer';
import { render, screen} from "@testing-library/react"
import TransactionRow from '../../components/TransactionRow';

it('', ()=>{
    //Arrange
    const item = {
        naam:"Nieuw item",
        datum: Date.now(),
        type:"Inkomsten",
        bedrag:33.00
    }
    
    render(
        <TransactionRow
            item = {item}
            editItem = {(item)=>{console.log("edit "+item.naam)}}
            handleDelete = {(item)=>{console.log("delete "+item.naam)}}
        ></TransactionRow>
    )    
})