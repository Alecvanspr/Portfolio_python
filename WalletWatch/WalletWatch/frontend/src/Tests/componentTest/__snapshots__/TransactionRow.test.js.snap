// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<tr>
  <td />
  <td>
    Nieuw item
  </td>
  <td>
    1661257928396
  </td>
  <td
    className="Inkomsten-text"
  >
    â‚¬
    33
  </td>
  <td>
    <button
      className="btn btn-secondary mr-2"
      onClick={[Function]}
    >
      Edit
    </button>
  </td>
  <td>
    <button
      className="btn btn-danger"
      onClick={[Function]}
    >
      Delete
    </button>
  </td>
</tr>
`;
